mineral-ui/mineral-ui
# [Website] normalize Guidelines pages layout

### Description
The component pages needed a common structure to help understanding of potential Mineral consumers. Now there will be an Title, followed by an Intro/Overview section, and also section navigation via a tab interface similar to the components pages.

- add lede with bigger font style
- modify flowconfig to enable markdown files to be loaded by name

## so the structure I'm proposing (as long as it suits us) would be:

This PR adds a table of contents for page sections which is injected after the intro paragraph. I'd suggest the section order could be informed by something like "Behavior" and "Usage" sections first-ish and then other sections specific to the page topic. This way we skew towards consumers with limited time to implement Mineral UI into their app and they get the most pragmatic information first.

```
# Title of the Guide page
This is an intro paragraph. This should probably only be 2-3 sentences, and shouldn't get into implementation details.

There could even be another paragraph here, it'll pick up the intro styles.

## Section Heading H2
A bunch of body copy

Resources (images, code blocks, examples, etc) underneath or mixed in with copy as needed.

... repeat as necessary

when we get several more pages, a Footer with Next and Resources
```

connects #355 

### Motivation and context
The Guidelines pages were a bit of a mish-mash, and there wasn't any clear structure to how a visitor should digest the page besides reading it from top to bottom. This PR adds some hierarchy to the page.

### How to test

1. `cd mineral-ui && npm start`
2. View the Getting Started and Theming pages.
3. There should be a title followed by an intro section
4. The navigation should correspond to all h2s defined in the markdown content for that page

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist
